pkg/apis/extensions/v1beta1/types.go

```diff
@@ -369,27 +369,28 @@ type DeploymentList struct {
 	Items []Deployment `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
 
-// TODO(madhusudancs): Uncomment while implementing DaemonSet updates.
-/* Commenting out for v1.2. We are planning to bring these types back with a more robust DaemonSet update implementation in v1.3, hence not deleting but just commenting the types out.
 type DaemonSetUpdateStrategy struct {
-	// Type of daemon set update. Only "RollingUpdate" is supported at this time. Default is RollingUpdate.
-// +optional
-	Type DaemonSetUpdateStrategyType `json:"type,omitempty"`
+	// Type of daemon set update. Can be "RollingUpdate" or "OnDelete".
+	// Default is OnDelete.
```

Comment:
I didn't read this PR, so ignore my comment if it does not make sense. There are cases that you may want to have `omitempty` but not `+optional`. If your intention is actually making a field optional, then add `+optional` otherwise we can discuss it more.